### PR TITLE
Feature/go formatting/sterling

### PR DESCRIPTION
**Pull request by sdreyer**
Includes go linting tests

### CODE REVIEW

**Review by Sami1309 — DISMISSED** [review]
Looks good, but go linter fails check. Maybe go linter should only check changed files, but that would guarantee less coverage in the beginning.

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -45,4 +45,51 @@ jobs:
...
+          go-version: 1.17
+
+      - name: Install grpc_tools
+        run: pip install grpcio-tools
+
+      - name: Install Protobuf
+        run: sudo snap install protobuf --classic
+
+      - name: Setup Proto
+        run: ./gen_grpc.sh
```
Comment (simba-git) [inline .github/workflows/coverage.yml:71]: Why?

Comment (sdreyer) [reply .github/workflows/coverage.yml:71]: Needed for the linter. Has import errors without it

Comment (simba-git) [reply .github/workflows/coverage.yml:71]: Ah I see, could you just add that in a comment just so its clear

**Review by simba-git — APPROVED** [review]
Just add one comment then good to go

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (sdreyer) [reply .github/workflows/coverage.yml:72]: Done

**Pull request merged**